### ISSUE TITLE
sendDocument Request timed out

### ISSUE THREAD

**Issue by hamidqp** [badfarm/zanzara#34]
Hi, Im using _Local Bot API Server_ for working with large files; but when i request for download or upload actually _telegram-bot-api_ is downloading that and until download completed it dosnt respond  to request (not like api.telegram.org) and makes this happend. 

`Failed to call Telegram Bot Api, method: sendDocument, params: {`
`    "document": {},`
`    "chat_id": 123456789,`
`    "parse_mode": "HTML"`
`}, reason: RuntimeException: Request timed out after 60 seconds in /vendor/react/http/src/Io/Transaction.php:130`
`Stack trace:`
`#0 /vendor/react/event-loop/src/Timer/Timers.php(96): React\Http\Io\Transaction->React\Http\Io\{closure}()`
`#1 /vendor/react/event-loop/src/StreamSelectLoop.php(184): React\EventLoop\Timer\Timers->tick()`
`#2 /vendor/badfarm/zanzara/src/Zanzara/Zanzara.php(90): React\EventLoop\StreamSelectLoop->run()`
`#3 /main.php(240): Zanzara\Zanzara->run()`
`#4 {main}`

wish i could contribute but i don't have enough knowledge :(

**Comment by cheeghi** [comment]
Hi, does your bot respond to [getMe](https://core.telegram.org/bots/api#getme) through your local server? Can you provide the code you use to send the document?

**Comment by hamidqp** [comment]
yes it works fine with normal requests, as i told when downloading or upload big files throw that.

btw here's part that upload files:
```
$ctx->sendDocument(new InputFile($name))->then(
    function () use ($pdf) {
        unlink($pdf);
    }
);
```

**Comment by cheeghi** [comment]
How big is your file? Does it happen with smaller files as well?

**Comment by cheeghi** [comment]
Anyways, I think the problem is that having a very big file, it takes time to send it to the server (probably more than 60 seconds, it depends on the file size and the connection) causing the connection going in timeout. I think it's useful to let the user specify a custom timeout for a request, of course it'll require a fix on the framework

**Comment by hamidqp** [comment]
yes, usually 500 Mg or higher. it would be nice to config that thanks :heart:

**Comment by cheeghi** [comment]
Solved in 0.6.4. To specify a timeout use:

```php
<?php

$bot = new Zanzara($_ENV['BOT_TOKEN']);

$bot->onCommand('start', function (Context $ctx) {
    $ctx->sendMessage('Hello', ['request_timeout' => 120]);
});
```
